setup/tue-functions.bash

```diff
@@ -230,29 +230,76 @@ complete -F __tue-git-clean-local _tue-git-clean-local
 #                                              SSH
 # ----------------------------------------------------------------------------------------------------
 
-function _git_https
+function _git_split_https
 {
     local url=$1
-    [[ "$url" == "https://"* ]] && echo "$url" && return 0
+    local web_address=${url#https://}
+    local domain_name=${web_address%%/*}
+    local repo_address=${web_address#*/}
+    repo_address=${repo_address%.git}
+    echo -e "$domain_name\n$repo_address"
+}
+export -f _git_split_https # otherwise not available in sourced files
 
-    local web_address=${url#git@}
+function _git_split_ssh
+{
+    local url=$1
+    local web_address=${url#ssh://}
+    web_address=${web_address#git@}
     local domain_name=${web_address%%:*}
     local repo_address=${web_address#*:}
+    repo_address=${repo_address%.git}
+    echo -e "$domain_name\n$repo_address"
+}
+export -f _git_split_ssh # otherwise not available in sourced files
+
+function _git_split_url
+{
+    local url=$1
+
+    local output
+    if [[ "$url" == *"@"* ]] # SSH
+    then
+        output=$(_git_split_ssh "$url")
+    else
+        output=$(_git_split_https "$url")
+    fi
+    # shellcheck disable=SC2086
+    echo $output
+}
+export -f _git_split_url # otherwise not available in sourced files
+
+function _git_https
+{
+    local url=$1
+    grep -q "^https://.*\.git$" <<< "$url" && echo "$url" && return 0
 
-    echo "https://$domain_name/$repo_address"
+    local output
+    output=$(_git_split_url "$url")
+
+    local array
+    read -r -a array <<< "$output"
+    local domain_name=${array[0]}
+    local repo_address=${array[1]}
+
+    echo "https://$domain_name/$repo_address.git"
 }
 export -f _git_https # otherwise not available in sourced files
 
 function _git_ssh
 {
     local url=$1
-    [[ "$url" == "git@"* ]] && echo "$url" && return 0
+    grep -q "^git@.*\.git$" <<< "$url" && echo "$url" && return 0
```

Comment:
I'm a bit against grep when the input is known to be a single string.

```suggestion
    [[ $url =~ ^git@.*\.git$ ]] && echo "$url" && return 0
```
This is better IMO.